src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -635,6 +641,13 @@ static void UpdateCurrentState(EvmState state, int pc, long gas, int stackHead)
             state.DataStackHead = stackHead;
         }
 
+        static void ApplyExternalState(EvmState state, out int pc, out long gas, out int stackHead)
```

Comment:
And this can be removed

Reply:
this method applies the changes that were done to the state from within the debugger context to the interpretation loop context

Reply:
it should be under `#if DEBUG` also why not to pass `pc`, `gas`, and `stackHead` by `ref` directly to the tracer so they can be modified there and not here?